darts/tests/utils/tabularization/test_create_lagged_training_data.py

```diff
@@ -550,15 +620,44 @@ def test_lagged_training_data_equal_freq(self, series_type: str):
                 output_chunk_shift,
             )
             # Number of observations should match number of feature times:
-            assert X.shape[0] == len(feats_times)
-            assert y.shape[0] == len(feats_times)
-            assert X.shape[0] == len(times[0])
-            assert y.shape[0] == len(times[0])
+            assert X.shape[0] == len(feats_times) == len(times[0])
+            assert y.shape[0] == len(feats_times) == len(times[0])
             # Check that outputs match:
             assert np.allclose(expected_X, X[:, :, 0])
             assert np.allclose(expected_y, y[:, :, 0])
             assert feats_times.equals(times[0])
 
+            # passing lags a dictionary
+            lags_as_dict = self.convert_lags_to_dict(
```

Comment:
It looks like in most of the tests, this code is duplicated from the non-dict lags case. We could add the logic as local function inside each test (including lagged feature extraction and checks), and then just that call it once with non-dict lags and once with dict lags.

Reply:
Done, let me know what you think